core/tauri/src/api/process/command.rs

```diff
@@ -277,9 +285,17 @@ impl Command {
     let guard_ = guard.clone();
     spawn(move || {
```

Comment:
we should probably refactor this copied logic into a simple function and call it once for stdout and stderr to prevent any future mistakes. from reading it it looks like the only differences are the reader and the output.

something like:
```rust
fn spawn_pipe_reader(reader: PipeReader, wrapper: impl Fn(String) -> CommandEvent) -> std::io::Result<()> {
  // -snip-
}
```

and then you should be able to do
```rust
spawn_pipe_reader(stdout_reader, CommandEvent::Stdout)?;
spawn_pipe_reader(stderr_reader, CommandEvent::Stderr)?;
```

Reply:
good idea, done